lapis/utility/monitor.py

```diff
@@ -1,19 +1,40 @@
+import copy
 from typing import Callable, TYPE_CHECKING
 
 import logging
+import logging.handlers
 
-from usim import each, Flag
-
-from lapis.cost import cobald_cost
+from cobald.monitor.format_json import JsonFormatter
+from cobald.monitor.format_line import LineProtocolFormatter
+from usim import each, Flag, time
 
 if TYPE_CHECKING:
     from lapis.simulator import Simulator
 
 sampling_required = Flag()
 
 
+class LoggingSocketHandler(logging.handlers.SocketHandler):
+    def makePickle(self, record):
+        return self.format(record).encode()
+
+
+class LoggingUDPSocketHandler(logging.handlers.DatagramHandler):
+    def makePickle(self, record):
+        return self.format(record).encode()
+
+
+class TimeFilter(logging.Filter):
+    """
+    py:class:`TimeFilter` takes care to modify the created timestamp of a log
+    record to be set to the current simulation time.
+    """
```

Comment:
Please shorten. No need to repeat the class name, and "takes care to modify ... to be set to" seems redundant.

How about "Dummy filter that sets the log record timestamp to the simulation time"?